doc/api.yml

```diff
@@ -0,0 +1,844 @@
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  ## Users resources
+  '/users':
+    get:
+      tags:
+        - User
+      summary: Get all users with criteria.
+      operationId: getUsers
+      parameters:
+        - name: page
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/Page'
+        - name: page_size
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: user_name
+          in: query
+          required: false
+          description: Filter user by first name, lastname or username.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: List of filtered users.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  '/users/{id}':
+    get:
+      tags:
+        - User
+      summary: Get user by identifier.
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: string
+      operationId: getUserById
+      responses:
+        '200':
+          description: The identified user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+    put:
+      tags:
+        - User
+      summary: Update user by identifier.
+      operationId: updateUserById
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: User to update.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        '201':
+          description: Updated user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+
+  ## Following resources
+  '/users/{id}/followers':
```

Comment:
L3 - Should be --> Get own followers with the request:
GET - /followers

check the reference picture in the Jira ticket to make the necessary modifications

Reply:
We want to be able to check a user followers too hence, I think /users/{id}/followers will work for both
`own followers` and `another user followers`